### ISSUE TITLE
False positive for MultilineOperationIndentation

### ISSUE THREAD

**Issue by taichi-ishitani** [rubocop-hq/rubocop#7094]
## Expected behavior

When enforce style of ` Layout/MultilineOperationIndentation` cop is `aligned` Rubocop should report no offences for multi-line operation without assignment like below.

```ruby
# frozen_string_literal: true

a = true
b = false

a &&
b
```

## Actual behavior

Rubocop reports `MultilineOperationIndentation` offence with following message:

```
sample.rb:7:1: C: Layout/MultilineOperationIndentation: Use 2 (not 0) spaces for indenting an expression spanning multiple lines.
b
^
```

## Steps to reproduce the problem

Unzip attached zip file and run command below.
[sample.zip](https://github.com/rubocop-hq/rubocop/files/3239619/sample.zip)

Then you can reproduce this issue.

```
$ rubocop -c sample.yml sample.rb
Inspecting 1 file
C

Offenses:

sample.rb:7:1: C: Layout/MultilineOperationIndentation: Use 2 (not 0) spaces for indenting an expression spanning multiple lines.
b
^

1 file inspected, 1 offense detected

```

## RuboCop version

```
$ rubocop -V
0.71.0 (using Parser 2.6.0.0, running on ruby 2.5.3 x64-mingw32)
```

**Comment by koic** [comment]
As far as I read the test code, this is the expected behavior common to both `EnforcedStyle: aligned (default)` and `EnforcedStyle: indented`. And I think this behavior makes sense.

```ruby
shared_examples 'common' do
  (snip)

  it 'registers an offense for no indentation of second line' do
    expect_offense(<<~RUBY)
      a +
      b
      ^ Use 2 (not 0) spaces for indenting an expression spanning multiple lines.
    RUBY
  end
```

https://github.com/rubocop-hq/rubocop/blob/v0.76.0/spec/rubocop/cop/layout/multiline_operation_indentation_spec.rb#L46-L52

I think example code and/or documentation should be added if the test code is correct.

**Comment by taichi-ishitani** [comment]
Hi @koic ,
Thank you for you comment and I understand this is the expected behavior.
I'm not sure whether or not this is the valid behavior because there are no sample code. So I open this ticket.
